File: python/a158.py
import sys
import math

for t in sys.stdin:
    t=int (t.replace('\n',''))
    for i in range (0,t):
        data = list(map(int,input().replace('\n','').split()))
        data.sort()
        max1=0
        for j in range(0,data.size()-1)
            for k in range(j+1,data.size()):
                max1=max(max1,math.gcd(data[k],data[j]))
        print (max1)
    

File: c++/a158.cpp
#include <iostream>
#include <sstream>
#include <cstring>
#include <algorithm>
#include <cmath>
#define _for(a,b,c) for(int (a)=(b);(a)<(c);(a)++)
using namespace std;

int gcd(int  a,int b){
	if (a%b == 0)
		return b;
	else 
		return gcd (b , a%b) ;
} 

int main()
{
	int t;
	string s;
	cin >> t;
	getline(cin,s);
	while (t--)
	{
		int a[101], m=0,max1=0;
		getline(cin,s);
		stringstream ss;
		ss<<s;
		while(ss>>a[m]) m++;
		sort(a,a+m);
		_for(i,0,m-1){
			_for(j,i+1,m)
				max1=max(max1,gcd(a[j],a[i]));
		}
		cout << max1 << endl;
	}
}
